docs/source/DevelopmentGuide.rst

```diff
@@ -0,0 +1,291 @@
+
+.. note::
+    If you don't want to use :code:`nox`, you can also run the tests directly using :code:`pytest`.
+
+    .. code-block:: console
+
+        (venv) $ pytest test/python
+
+Python Code Formatting and Linting
+----------------------------------
+
+The Python code is formatted and linted using a collection of `pre-commit hooks <https://pre-commit.com/>`_.
+This collection includes:
+
+- `black <https://black.readthedocs.io/en/stable/>`_ -- a code formatter that automatically formats Python code according to the `PEP 8 style guide <https://www.python.org/dev/peps/pep-0008/>`_
+- `flake8 <https://flake8.pycqa.org/en/latest/>`_ -- a linter that checks for common mistakes in Python code
+- `isort <https://pycqa.github.io/isort/>`_ -- a tool that automatically sorts Python imports according to the `PEP 8 style guide <https://www.python.org/dev/peps/pep-0008/>`_
+- `mypy <http://mypy-lang.org/>`_ -- a static type checker for Python code
+- `pyupgrade <https://github.com/asottile/pyupgrade>`_ -- a tool that automatically upgrades Python syntax to a newer version
+
+There are two ways of using these hooks:
+
+- You can install the hooks manually by running :code:`pre-commit install` in the project root directory.
+  This will install the hooks in the :code:`.git/hooks` directory of the repository.
+  The hooks will then be executed automatically when committing changes.
+- You can use the :code:`nox` session :code:`lint` to run the hooks manually.
+
+    .. code-block:: console
+
+        (venv) $ nox -rs lint
+
+    .. note::
+        If you don't want to use :code:`nox`, you can also run the hooks directly using :code:`pre-commit`.
+
+        .. code-block:: console
+
+            (venv) $ pre-commit run --all-files
+
+In addition to the pre-commit hooks, the Python code is also type checked by `mypy <http://mypy-lang.org/>`_.
+This is done by the :code:`nox` session :code:`mypy`.
+
+    .. code-block:: console
+
+        (venv) $ nox -rs mypy
+
+Working on the Documentation
+############################
+
+The documentation is written in `reStructuredText <https://docutils.sourceforge.io/rst.html>`_ and built using `Sphinx <https://www.sphinx-doc.org/en/master/>`_.
+The documentation source files can be found in the :code:`docs/source` directory.
+You can build the documentation using the :code:`nox` session :code:`docs`.
+
+    .. code-block:: console
+
+        (venv) $ nox -rs docs
+
+This will install all dependencies for building the documentation in an isolated environment, build the Python package, and then build the documentation.
+The session also provides a convenient option to automatically serve the docs on a local web server. Running
+
+    .. code-block:: console
+
+        (venv) $ nox -rs docs -- serve
+
+will start a local web server on port 8000 and provide a link to open the documentation in your browser.
+
+To build the documentation without (re-)installing the Python package, you can use the :code:`skip-install` flag.
```

Comment:
These `skip-install` options are not yet implemented in the noxfile of this repository. You can adapt a corresponding solution from the version in QCEC.

Reply:
In the noxfile.py I need to to include the coverage and mypy right?

Reply:
No need to do that here. I'll include them in the Python test CI setup (where they are actively used). See the comment above.

Reply:
done 👍🏻